### PR TITLE
Refactor/fix current index row

### LINKED ISSUE #613: Wrong current index row
The current index row selection implemented in #561 is flawed. Consider following scenario:

**Index**
```
1  First Section
  1.1  Subsection One
  1.2  Subsection Two
2  Second Section
```

Scrolling the document right before `2  Second Section`, the "current index row" is then chosen as `1  First Section`.

### PR DESCRIPTION
**Pull request by jdujava**
Refactor and simplify by remembering the last index path with `index path page number` <= `current page`.

Fix #613.

### CODE REVIEW

File: zathura/utils.c [hunk 1]
```diff
@@ -108,42 +108,27 @@ void document_index_build(girara_session_t* session, GtkTreeModel* model, GtkTre
   }
 }
 
-gboolean _for_each_func(GtkTreeModel* model, GtkTreePath* path, GtkTreeIter* iter, gpointer data) {
+gboolean _update_current_index_path(GtkTreeModel* model, GtkTreePath* path, GtkTreeIter* iter, gpointer data) {
```
Comment (sebastinas) [inline zathura/utils.c:111]: This function could be static.

File: zathura/utils.c [hunk 2]
```diff
@@ -108,42 +108,27 @@ void document_index_build(girara_session_t* session, GtkTreeModel* model, GtkTre
...
-      path = gtk_tree_model_get_path(model, &parent_iter);
-    }
-    gtk_tree_view_expand_to_path(tree_view, path);
-    gtk_tree_view_set_cursor(tree_view, path, NULL, FALSE);
-    gtk_tree_view_scroll_to_cell(tree_view, path, NULL, TRUE, 0.5, 0.0);
-
-    return TRUE;
+    return TRUE; // terminate foreach loop
   }
+  zathura->global.current_index_path = gtk_tree_path_copy(path);
```
Comment (sebastinas) [inline zathura/utils.c:120]: Does this need to be freed at some point?

Comment (jdujava) [reply zathura/utils.c:120]: Thanks, hopefully now it is OK.

Comment (sebastinas) [reply zathura/utils.c:120]: I would also expect a `gtk_tree_path_free` here if `zathura->global.current_index_path` is not `NULL`.

Comment (jdujava) [reply zathura/utils.c:121]: Yeah, of course..

**Comment by jdujava** [comment]
Pushed even better (in my opinion) refactor.

**Pull request merged**